### ISSUE TITLE
datascience-notebook: Julia kernel not available for selection 

### ISSUE THREAD

**Issue by pdmct** [jupyter/docker-stacks#30]
Julia kernel doesn't appear in the list of available kernels in the Jupyter interface. (Only Python2,3 and R are listed)

**Comment by pdmct** [comment]
I tried manually updating the kernelspec by adding to Dockerfile:

`RUN $CONDA_DIR/envs/python2/bin/python     $CONDA_DIR/envs/python2/bin/ipython     kernelspec install --replace --name ijulia /root/.ipython/kernels/julia-0.3/`

Which gets Julia to show up in the available kernel list -- but kernel doesn't run/crashes when I try to do anything.

**Comment by parente** [comment]
Which docker image tag did you pull? latest?

**Comment by parente** [comment]
I see it too. Looks like another problem with the kernel spec installing into the active user directory (root) when the dockerfile is running instead of system wide. This broke when we moved the jovyan user creation out of the Dockerfile and into the start script to give control over the UID (PR #16)

The possible fix for issue #25 will probably fix regression too.

**Comment by parente** [comment]
With PR #31 and this change in the datascience-notebook/Dockerfile everything works again:

```
USER jovyan

# Install IJulia packages as jovyan and then move the kernelspec out
# to the system share location. Avoids problems with runtime UID change not 
# taking effect properly on the .local folder in the jovyan home dir.
RUN julia -e 'Pkg.add("IJulia")' && \
    mv /home/$NB_USER/.local/share/jupyter $CONDA_DIR/share/ && \
    chmod -R go+rx $CONDA_DIR/share/jupyter && \
    rm -rf /home/$NB_USER/.local/share
RUN julia -e 'Pkg.add("Gadfly")' && julia -e 'Pkg.add("RDatasets")'

USER root
```

PR #31 needs more testing before merging however.

**Comment by parente** [comment]
IJulia just had a regression against Julie 0.3 that will prevent builds even after the PR mentioned above goes into master here. 

https://github.com/JuliaLang/IJulia.jl/commit/e01b2e44f18b2f1dbcc167273949dd2ac8258dea#commitcomment-13197892

**Comment by parente** [comment]
Now fixed in the `latest` and `4.0` tags on Docker Hub.
